src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,206 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size = 0;
+    private  int midIndex = size / 2;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirstNode(value);
+        } else {
+            addNoteToEnd(value);
```

Comment:
The method name `addNoteToEnd` seems to be a typo and should be `addNodeToEnd` to correctly reflect its functionality.